Add edge-21.1.1 upgrade notes (#887)
linkerd/website: linkerd.io/content/2/tasks/upgrade.md

version. If you have any questions, feel free to raise them at the #linkerd2
channel in the [Linkerd slack](https://slack.linkerd.io/).

## Upgrade notice: edge-21.1.1

### Visualization components moved to Linkerd-Viz extension

All of the Linkerd control plane components related to visibility (including
Prometheus, Grafana, Web, and Tap) have been removed from the main Linkerd
control plane and moved into the Linkerd-Viz extension. This means that when you
upgrade to edge-21.1.1, these components will be removed from your cluster and
you will not be able to run commands such as `linkerd stat` or `linkerd
dashboard`. To restore this functionality, you must install the Linkerd-viz
extension by running `linkerd viz install | kubectl apply -f -`.

```bash
# Upgrade the control plane (this will remove viz components).

> linkerd upgrade | kubectl apply --prune -l linkerd.io/control-plane-ns=linkerd -f -

Reviewer: should this have the more verbose prune instructions?

# Install the Linkerd-Viz extension to restore viz functionality.
linkerd viz install | kubectl apply -f -
```

Helm users should note that configuration values related to these visibility
components have moved to the Linkerd-Viz chart. Please update any values
overrides you have and use these updated overrides when upgrading the Linkerd
chart or installing the Linkerd-Viz chart. See below for a complete list of
values which have moved.

```bash
helm repo update
# Upgrade the control plane (this will remove viz components).

> helm upgrade linkerd2 linkerd/linkerd2 --reset-values -f values.yaml --atomic

Reviewer 2: Given we're upgrading to an edge, this should read instead:
```suggestion
helm upgrade linkerd2 linkerd-edge/linkerd2 --reset-values -f values.yaml --atomic
```

Reviewer 2: Also, what do you think about mentioning using `helm get values` prior to that reset in order to get the previous values?

Author: that's not really related to this edge in particular.  I also want to update the general helm upgrade docs in a separate PR.

# Install the Linkerd-Viz extension to restore viz functionality.

> helm install linkerd2-viz linkerd/linkerd2-viz -f viz-values.yaml

Reviewer 2: ```suggestion
helm install linkerd2-viz linkerd-edge/linkerd2-viz -f viz-values.yaml
```

```

The following values were removed from the Linkerd2 chart. Most of the removed
values have been moved to the Linkerd-Viz chart or the Linkerd-Jaeger chart.

* `dashboard.replicas` moved to Linkerd-Viz as `dashboard.replicas`
* `tap` moved to Linkerd-Viz as `tap`
* `tapResources` moved to Linkerd-Viz as `tap.resources`
* `tapProxyResources` removed
* `webImage` moved to Linkerd-Viz as `dashboard.image`
* `webResources` moved to Linkerd-Viz as `dashboard.resources`
* `webProxyResources` removed
* `grafana` moved to Linkerd-Viz as `grafana`

> * `grafnaa.proxy` removed

Reviewer 2: ```suggestion
* `grafana.proxy` removed
```

version. If you have any questions, feel free to raise them at the #linkerd2
channel in the [Linkerd slack](https://slack.linkerd.io/).

## Upgrade notice: stable-2.10.0

### Visualization components moved to Linkerd-Viz extension

All of the Linkerd control plane components related to visibility (including
Prometheus, Grafana, Web, and Tap) have been removed from the main Linkerd
control plane and moved into the Linkerd-Viz extension. This means that when you

> upgrade to edge-21.1.1, these components will be removed from your cluster and

Reviewer 2: ```suggestion
upgrade to stable-2.10.0, these components will be removed from your cluster and
```

you will not be able to run commands such as `linkerd stat` or `linkerd
dashboard`. To restore this functionality, you must install the Linkerd-Viz

> extension by running `linkerd viz install | kubectl apply -f -`.

Reviewer 2: "... and then invoke those commands through `linkerd viz stat`, `linkerd viz dashboard`, etc."

# Install the Linkerd-Viz extension to restore viz functionality.
helm install linkerd2-viz linkerd-edge/linkerd2-viz -f viz-values.yaml
```

The following values were removed from the Linkerd2 chart. Most of the removed
values have been moved to the Linkerd-Viz chart or the Linkerd-Jaeger chart.

- `dashboard.replicas` moved to Linkerd-Viz as `dashboard.replicas`
- `tap` moved to Linkerd-Viz as `tap`
- `tapResources` moved to Linkerd-Viz as `tap.resources`
- `tapProxyResources` removed
- `webImage` moved to Linkerd-Viz as `dashboard.image`
- `webResources` moved to Linkerd-Viz as `dashboard.resources`
- `webProxyResources` removed
- `grafana` moved to Linkerd-Viz as `grafana`
- `grafana.proxy` removed
- `prometheus` moved to Linkerd-Viz as `prometheus`
- `prometheus.proxy` removed
- `global.proxy.trace.collectorSvcAddr` moved to Linkerd-Jaeger as `webhook.collectorSvcAddr`
- `global.proxy.trace.collectorSvcAccount` moved to Linkerd-Jaeger as `webhook.collectorSvcAccount`
- `tracing.enabled` removed
- `tracing.collector` moved to Linkerd-Jaeger as `collector`
- `tracing.jaeger` moved to Linkerd-Jaeger as `jaeger`

> 

Reviewer 2: "... also please note the `global` scope from the Linkerd2 chart values has been dropped, moving the config values underneath it into the root scope. Any values you had customized there will need to be migrated; in particular `identityTrustAnchorsPEM` in order to conserve the value you set during install."

extension by running `linkerd viz install | kubectl apply -f -` and then invoke
those commands through `linkerd viz stat`, `linkerd viz dashboard`, etc.

```bash
# Upgrade the control plane (this will remove viz components).
linkerd upgrade | kubectl apply --prune -l linkerd.io/control-plane-ns=linkerd -f -
# Prune cluster-scoped resources
linkerd upgrade | kubectl apply --prune -l linkerd.io/control-plane-ns=linkerd \
  --prune-whitelist=rbac.authorization.k8s.io/v1/clusterrole \
  --prune-whitelist=rbac.authorization.k8s.io/v1/clusterrolebinding \
  --prune-whitelist=apiregistration.k8s.io/v1/apiservice -f -
# Install the Linkerd-Viz extension to restore viz functionality.
linkerd viz install | kubectl apply -f -
```

Helm users should note that configuration values related to these visibility
components have moved to the Linkerd-Viz chart. Please update any values
overrides you have and use these updated overrides when upgrading the Linkerd
chart or installing the Linkerd-Viz chart. See below for a complete list of
values which have moved.

```bash
helm repo update
# Upgrade the control plane (this will remove viz components).

> helm upgrade linkerd2 linkerd-edge/linkerd2 --reset-values -f values.yaml --atomic
> # Install the Linkerd-Viz extension to restore viz functionality.
> helm install linkerd2-viz linkerd-edge/linkerd2-viz -f viz-values.yaml

Reviewer 2: Since this are the upgrade notes for a stable version, we should use the stable repo
```suggestion
helm upgrade linkerd2 linkerd/linkerd2 --reset-values -f values.yaml --atomic
# Install the Linkerd-Viz extension to restore viz functionality.
helm install linkerd2-viz linkerd/linkerd2-viz -f viz-values.yaml
```

```

The following values were removed from the Linkerd2 chart. Most of the removed
values have been moved to the Linkerd-Viz chart or the Linkerd-Jaeger chart.

- `dashboard.replicas` moved to Linkerd-Viz as `dashboard.replicas`
- `tap` moved to Linkerd-Viz as `tap`
- `tapResources` moved to Linkerd-Viz as `tap.resources`

> - `tapProxyResources` removed

Reviewer 2: ```suggestion
- `tapProxyResources` moved to Linkerd-Viz as `tap.proxy.resources`
```

- `webImage` moved to Linkerd-Viz as `dashboard.image`
- `webResources` moved to Linkerd-Viz as `dashboard.resources`

> - `webProxyResources` removed

Reviewer 2: ```suggestion
- `webProxyResources` moved to Linkerd-Viz as `dashboard.proxy.resources`
```

- `grafana` moved to Linkerd-Viz as `grafana`

> - `grafana.proxy` removed

Reviewer 2: ```suggestion
- `grafana.proxy` moved to Linkerd-Viz as `grafana.proxy`
```

- `prometheus` moved to Linkerd-Viz as `prometheus`

> - `prometheus.proxy` removed

Reviewer 2: This lives in Linkerd-Viz, as-is
```suggestion
```
